barretenberg/cpp/src/barretenberg/plonk_honk_shared/composer/composer_lib.hpp

```diff
@@ -25,7 +25,10 @@ void construct_lookup_table_polynomials(const RefArray<typename Flavor::Polynomi
     const size_t tables_size = circuit.get_tables_size();
     ASSERT(tables_size <= MAX_LOOKUP_TABLES_SIZE); // if false, may need to increase constant
     ASSERT(dyadic_circuit_size > tables_size + additional_offset);
-    size_t offset = dyadic_circuit_size - tables_size - additional_offset;
+    size_t offset = circuit.blocks.lookup.trace_offset;
+    if constexpr (IsPlonkFlavor<Flavor>) {
```

Comment:
this is the right thing to do but ugh. Plonk.

Reply:
getting closer and closer to removing it is what one would hope